activerecord/lib/active_record/relation/merger.rb

```diff
@@ -23,7 +23,7 @@ def merge #:nodoc:
       # build a relation to merge in rather than directly merging
       # the values.
       def other
-        other = Relation.create(relation.klass, relation.table, relation.predicate_builder)
```

Comment:
I think it is safer and more correct to copy the table and the predicate_builder of the relation here instead of trying to get from the klass

Reply:
Oh, good point. I hadn't planned to change the existing behavior. I probably need more ☕️ . Will update now.